dripline/core/request_sender.py

```diff
@@ -0,0 +1,132 @@
+            confirm_retcodes: bool, optional  
+                If True, and if a reply is received with retcode != 0, raises an exception -- default is True          
+        '''
+        self.sender = sender
+        self._confirm_retcode = confirm_retcodes
+        self.timeout_s = timeout_s
+        self._receiver = Receiver()
+
+
+    def _check_lockout_key(self, lockout_key:str | uuid.UUID =None):
+        nilkey = uuid.UUID('00000000-0000-0000-0000-000000000000')
+        if lockout_key is None:
+            return nilkey
+        if type(lockout_key) == uuid.UUID:
+            return lockout_key
+        try:
+            return uuid.UUID(lockout_key)
+        except ValueError:
+            logger.warning("Lockout Key '{}' is not properly uuid formatted. Defaulting to nilkey. ".format(lockout_key))
+            return nilkey
+
+
+    def _send_request(self, msgop, target, specifier=None, payload=None, timeout=None, lockout_key=None):
+        '''
+        internal helper method to standardize sending request messages
+        '''
+        a_specifier = specifier if specifier is not None else ""
+        a_request = MsgRequest.create(payload=scarab.to_param(payload), msg_op=msgop, routing_key=target, specifier=a_specifier)
+        a_request.lockout_key = self._check_lockout_key(lockout_key)
+        receive_reply = self.sender.send(a_request)
+        if not receive_reply.successful_send:
+            raise DriplineError('unable to send request')
+        return receive_reply
+
+    def _receive_reply(self, reply_pkg, timeout_s):
+        '''
+        internal helper method to standardize receiving reply messages
+        '''
+        sig_handler = scarab.SignalHandler()
+        sig_handler.add_cancelable(self._receiver)
+        result = self._receiver.wait_for_reply(reply_pkg, timeout_s * 1000) # receiver expects ms
+        sig_handler.remove_cancelable(self._receiver)
+        return result.payload.to_python()
+
+    def get(self, endpoint: str, specifier: str=None, lockout_key=None, timeout_s: int=0) -> MsgReply:
+        '''
+        Send a get request to an endpoint and return the reply message.
+
+        Parameters
+        ----------
+            endpoint: str
+                Routing key to which the request should be sent.
+            specifier: str, optional
+                Specifier to add to the request, if needed.
+            timeout_s: int | float, optional
+                Maximum time to wait for a reply in seconds (default is 0)
+                A timeout of 0 seconds means no timeout will be used.
+        '''
+        reply_pkg = self._send_request( msgop=op_t.get, target=endpoint, specifier=specifier, lockout_key=lockout_key )
+        result = self._receive_reply( reply_pkg, timeout_s )
+        return result
+
+    def set(self, endpoint: str, value: str | int | float | bool, specifier: str=None, lockout_key=None, timeout_s: int | float=0) -> MsgReply:
```

Comment:
As a general improvement, not an issue with anything you did, can you please add a type to `lockout_key`?  It can be `lockout_key: str | uuid.UUID`, I believe.